packages/lesswrong/components/users/DialogueMatchingPage.tsx

```diff
@@ -542,197 +888,143 @@ const MessageButton: React.FC<{
   );
 };
 
-type UserTableProps = {
-  users: any[]; // replace with the correct type
-  classes: ClassesType;
-  gridClassName: string,
-  currentUser: UsersCurrent;
-  userDialogueChecks: DialogueCheckInfo[];
-  loadingNewDialogue: boolean;
-  createDialogue: (title: string, participants: string[]) => void;
-  showBio?: boolean;
-  showKarma?: boolean;
-  showAgreement?: boolean;
-  showPostsYouveRead?: boolean;
-  showFrequentCommentedTopics?: boolean;
-};
 
-const UserTable: React.FC<UserTableProps> = ({
-  users,
-  classes,
-  gridClassName,
-  currentUser,
-  userDialogueChecks,
-  loadingNewDialogue,
-  createDialogue,
-  showBio,
-  showKarma,
-  showAgreement,
-  showPostsYouveRead,
-  showFrequentCommentedTopics,
-}) => {
+const DialogueUserRow = <V extends boolean>(props: DialogueUserRowProps<V> & { classes: ClassesType }): JSX.Element => {
+  const { targetUser, checkId, userIsChecked, userIsMatched, classes, currentUser, showKarma, showAgreement, showBio, showFrequentCommentedTopics, showPostsYouveRead } = props;
+  const { UsersName, DialogueCheckBox, MessageButton, MatchDialogueButton } = Components;
+
+  return <React.Fragment key={`${targetUser._id}_other`}>
+    <DialogueCheckBox
+      targetUserId={targetUser._id}
+      targetUserDisplayName={targetUser.displayName}
+      checkId={checkId}
+      isChecked={userIsChecked}
+      isMatched={userIsMatched}
+    />
+    <UsersName
+      className={classes.displayName}
+      documentId={targetUser._id}
+      simple={false} />
+    <MessageButton
+      targetUserId={targetUser._id}
+      currentUser={currentUser}
+    />
+    <MatchDialogueButton
+      isMatched={userIsMatched}
+      checkId={checkId}
+      targetUserId={targetUser._id}
+      targetUserDisplayName={targetUser.displayName}
+      currentUser={currentUser}
+    />
+    {showKarma && <div className={classes.centeredText}> {targetUser.total_power} </div>}
+    {showAgreement && <div className={classes.centeredText}> {targetUser.total_agreement} </div>}
+    {showBio && <UserBio
+      key={targetUser._id}
+      classes={classes}
+      userId={targetUser._id} />}
+    {showFrequentCommentedTopics && <UserTopTags
+      classes={classes}
+      targetUserId={targetUser._id} />}
+    {showPostsYouveRead && <UserPostsYouveRead
+      classes={classes}
+      targetUserId={targetUser._id}
+      limit={8} />}
+  </React.Fragment>;
+}
+
+const UserTable = <V extends boolean>(props: UserTableProps<V>) => {
+  const {
+    users,
+    classes,
+    gridClassName,
+    userDialogueChecks,
+    showHeaders,
+    ...rest
```

Comment:
rename to something better like `remainingProps`

Reply:
(robert wrote that)

Reply:
(Robert told me to leave a note saying change to something like `remainingProps` ;)